File: ruby-tests/rectangle.rb
class Rectangle
  attr_reader :length, :width

  def initialize(length, width)
    @length = length
    @width = width
  end

  def circumference
    2 * (@length + @width)
  end

  def area
    @length * @width
  end

  def to_s
    "Rectangle(length: #{@length}, width: #{@width})"
  end
end

class Square < Rectangle
  def initialize(side_length)
    super(side_length, side_length)
  end

  def to_s
    "Square(side_length: #{@length})"
  end
end

File: pytest-tests/rectangle.py
class Rectangle:
    """
    Represents a 2D rectangle with a length and width.
    Provides methods to compute circumference and area.
    """
    def __init__(self, length: float, width: float):
        self.length = length
        self.width = width

    def circumference(self) -> float:
        """Returns the circumference of the rectangle."""
        return 2 * (self.length + self.width)

    def area(self) -> float:
        """Returns the area of the rectangle."""
        return self.length * self.width


class Square(Rectangle):
    """
    Represents a 2D square, which is a special type of rectangle
    where the length and width are equal.
    """
    def __init__(self, side_length: float):
        super().__init__(side_length, side_length)
